src/hooks/useArNSContract/useArNSContract.tsx

```diff
@@ -0,0 +1,49 @@
+import { LocalFileSystemDataProvider } from '../../services/arweave/LocalFilesystemDataProvider';
+import { useStateValue } from '../../state/state';
+import { useState, useEffect } from 'react';
+
+const localContractAddress = 'bLAgYxAdX2Ry-nt6aH2ixgvJXbpsEYm28NgJgyqfs-U';
+
+export default function useArNSContract() {
+  const [{ arnsSourceContract }, dispatch] = useStateValue();
+  const [sendingContractState, setSendingContractState] = useState(false);
+
+
+
+    useEffect(() => {
+      dispatchNewContractState();
+    }, []);
+    async function dispatchNewContractState(): Promise<void> {
+      try {
+        const localProvider = new LocalFileSystemDataProvider();
+      if (sendingContractState) {
+        return;
+      }
+      setSendingContractState(true);
+
+      const localState = await localProvider.getContractState(
+        localContractAddress,
+      );
+      if (!localState) {
+        throw Error('ArNS contract state is empty');
+      }
+      console.log(localState);
+      dispatch({
+        type: 'setArnsContractState',
+        payload: { records: localState.records },
+      });
+
+      setTimeout(() => {
+        // 0.1 second delay before dispatching another calculation to prevent UI jitter
+        setSendingContractState(false);
+      }, 100);
+    }
+    catch (error) {
+      console.log(
+        `error in setting contract state to GlobalState provider >>>`,
```

Comment:
this is super nit-picky, but I'm a strong proponent for sentences in error messages. 

```suggestion
        `Error in setting contract state.`,
```